HPCC-23479 add *eclmod files for LN plugins in HPCC-Platform

xwang2713 opened the pull request:
All *eclmod files, created from LN plugins header files are under plugins/proxies.
I also include the first one lib_addrcleanlib.eclmod

Install part are added in CMakeFiles.txt

Signed-off-by: Xiaoming Wang <xiaoming.wang@lexisnexis.com>

@ghalliday please review


## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [ ] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
<!-- Please describe how this change has been tested.-->
http://10.240.32.242/builds/custom/JIRA/HPCC-23479/

Will ask ECL developers and tester to test the builds

<!-- Thank you for taking the time to submit this pull request and to answer all of the above-->

xwang2713 committed:
HPCC-23479 add *eclmod files for LN plugins in HPCC-Platform

HPCCSmoketest commented:
Process of PR-13390, label: HPCC-23479-eclmod-platform starts now.
The reason of this test is: Did not build yet.
Commit ID: b3d17a60e1f53f364f54d88999dc7d85b5b479e1
Estimated completion time is ~0.9 hour(s)

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-17.ca-central-1.compute.internal
Sha: b3d17a60e1f53f364f54d88999dc7d85b5b479e1
Build: success
Milestone:Install hpccsystems-platform-community_7.6.31-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |116 |116 |0 |0 |0 |35 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |26 sec  (00:00:26) |
|setup (thor) |11 |11 |0 |45 sec  (00:00:45) |
|setup (roxie) |11 |11 |0 |19 sec  (00:00:19) |
|test (hthor) |868 |868 |0 |463 sec  (00:07:43) |
|test (thor) |807 |807 |0 |887 sec  (00:14:47) |
|test (roxie) |949 |949 |0 |585 sec  (00:09:45) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    7 sec (00:00:07) |  638 sec (00:10:38) |  110 sec (00:01:50) |   20 sec (00:00:20) |   17 sec (00:00:17) | 2269 sec (00:37:49) |   19 sec (00:00:19) | 3080 sec (00:51:20) |

ghalliday commented on plugins/proxies/lib_date.eclmod:
```diff
@@ -0,0 +1,91 @@
+/*##############################################################################
```
This file does not relate to one of the changes from LN.  Was it released by mistake/

xwang2713 replied:
I saw this in current <LN>/plugins/ directory. Thought it should be copied to HPCC-Platform. If not I can remove it from HPCC-Platform

ghalliday replied:
@richardkchapman not sure what to do about this one.  I don't think we want this plugin included in the system - it is really a legacy LN module, superseded by Std.Date etc.

Should we include all the LN plugins in a separate directory (e.g., plugins/eeproxies)?

It should be deleted from LN if it is being moved to this repo.

ghalliday commented on plugins/proxies/lib_dmslib.eclmod:
```diff
@@ -0,0 +1,22 @@
+/*##############################################################################
```
@richardkchapman should we just delete this plugin.  I don't think it is used by any queries.  Possibly remove in 7.8.0 instead?

ghalliday commented:
moxiekeylib appears to be missing.

ghalliday commented on plugins/proxies/lib_hktlib.eclmod:
```diff
@@ -0,0 +1,26 @@
+    limitations under the License.
+############################################################################## */
+
+/* Proxy plugin definition for (EE only) hkt library */
```
Another candidate to remove later

ghalliday commented on plugins/proxies/lib_namelib.eclmod:
```diff
@@ -0,0 +1,29 @@
+  string20  TokenToFirstName(const string token) : c, pure, entrypoint='nameTokenToFirstName';
+  unsigned4 TokenToLength(const string name) : c, pure, entrypoint='nameTokenToLength';
+END;
+export fnstring := TYPE
```
This type could be deleted in 7.8.  I'm not convinced the plugin is used at all.

ghalliday reviewed (commented):
@xwang2713 a couple of comments.  The comments about removing plugins should be dealt with in a separate PR

ghalliday commented on plugins/proxies/CMakeLists.txt:
```diff
@@ -20,6 +20,22 @@ install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/lib_metaphone.ecllib DESTINATION plu
+install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/lib_ofaclib.eclmod DESTINATION plugins COMPONENT Runtime)
+install ( FILES ${CMAKE_CURRENT_SOURCE_DIR}/lib_ziplib.eclmod DESTINATION plugins COMPONENT Runtime)
+
+
```
trivial: extra blank line

xwang2713 commented:
> moxiekeylib appears to be missing.
There is lib_keylib.ecllib file which has the content I want to create for *.eclmod. So I didn't create lib_moxiekeylib.eclmod

ghalliday commented:
@xwang2713 see follow on notes in the jira.

xwang2713 committed:
HPCC-23479 create proxies/*.eclmod from LN plugins
Install *.eclmod based on cmake options USE_PROXIES, USE_EEPROXISE and
USE_ECLCCPROXIES.

HPCCSmoketest commented:
Process of PR-13390, label: HPCC-23479-eclmod-platform starts now.
The reason of this test is: New commit.
Commit ID: e14ca56f501ff780c4fcec6f45913eefc2a085d0
Estimated completion time is ~0.9 hour(s)
centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal

xwang2713 commented:
Re-implemented per instruction in the JIRA. 
New PR: https://github.com/hpcc-systems/HPCC-Platform/pull/13497

HPCCSmoketest commented:
Automated Smoketest: :white_check_mark:
OS: centos 7.7.1908 (Linux 3.10.0-957.1.3.el7.x86_64)
 GCC: gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Host: ip-10-20-0-116.ca-central-1.compute.internal
Sha: e14ca56f501ff780c4fcec6f45913eefc2a085d0
Build: success
ECL Watch:  Rebuilding Site
Install warning(s): 
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/watchpack/node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {os:darwin,arch:any} (current: {os:linux,arch:x64})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.9 (node_modules/fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.9: wanted {os:darwin,arch:any} (current: {os:linux,arch:x64})

Rebuild: success
Milestone:Install hpccsystems-platform-community_7.6.33-closedown0.el7.x86_64.rpm
HPCC Start: OK

Unit tests result:

| Test | total | passed | failed | errors | timeout | elaps | 
|---|---|---|---|---|---|---|
|unittest |116 |116 |0 |0 |0 |40 sec |
|wutoolTest(Dali) |19 |19 |0 |0 |0 |2 sec |
|wutoolTest(Cassandra) |19 |19 |0 |0 |0 |9 sec |



Regression test result:

| phase | total | pass | fail | elaps | 
|---|---|---|---|---|
|setup (hthor) |11 |11 |0 |27 sec  (00:00:27) |
|setup (thor) |11 |11 |0 |44 sec  (00:00:44) |
|setup (roxie) |11 |11 |0 |20 sec  (00:00:20) |
|test (hthor) |868 |868 |0 |570 sec  (00:09:30) |
|test (thor) |807 |807 |0 |963 sec  (00:16:03) |
|test (roxie) |949 |949 |0 |713 sec  (00:11:53) |


HPCC Stop: OK
HPCC Uninstall: OK
Time stats:

| Prep time | Build time | Package time | Install time | Start time | Test time | Stop time | Summary | 
|---|---|---|---|---|---|---|---|
|    8 sec (00:00:08) |  679 sec (00:11:19) |  246 sec (00:04:06) |   23 sec (00:00:23) |   18 sec (00:00:18) | 2564 sec (00:42:44) |   17 sec (00:00:17) | 3555 sec (00:59:15) |
